### PR TITLE
Read-only service accounts for automation and CI/CD

### PR DESCRIPTION
**Pull request by ludoo**
WIP, design document in this PR has the rationale and explanation of changes.

This PR also

- adds a custom roles factory to the organization and project modules
- refactors the factory variables in organization, folder and project modules so they use our latest standard
- moves FAST stage 0 custom roles to YAML files passed to the organization module's factory
- removes the (now stale) `custom_role_names` variable from FAST stage 0
- refactors the FAST CI/CD workflows

Test errors will be expected until this has been completed.

### CODE REVIEW

**Comment by ludoo** [comment]
This is now ready, barring some IAM tweaks for the stage 2 and 3 CI/CD read-only service accounts, which I will test today. It can be reviewed, additional roles will be added before merging but won't change the PR structure.

**Comment by ludoo** [comment]
This is now ready for review. The second set of service accounts is still missing from multitenant stages, but I will add it once the approach has been validated. CI/CD tested with actual repositories for stages 0 and 1.

File: fast/assets/templates/workflow-github.yaml [hunk 1]
```diff
@@ -46,60 +46,82 @@ jobs:
...
         name: Authenticate to Google Cloud
-        uses: google-github-actions/auth@v0
+        uses: google-github-actions/auth@v2
         with:
-          workload_identity_provider: $${{ env.FAST_WIF_PROVIDER }}
-          service_account: $${{ env.FAST_SERVICE_ACCOUNT }}
-          access_token_lifetime: 3600s
+          workload_identity_provider: $${{env.FAST_WIF_PROVIDER}}
+          service_account: $${{env.service_account}}
+          access_token_lifetime: 900s
```
Comment (juliocc) [inline fast/assets/templates/workflow-github.yaml:81]: nit: 15 minutes sounds a bit on the low end for larger stages

File: fast/assets/templates/workflow-gitlab.yaml [hunk 2]
```diff
@@ -12,113 +12,88 @@
...
-    - echo "$${GITLAB_TOKEN}" > token.txt
     - |
       gcloud iam workload-identity-pools create-cred-config \
-        $${FAST_WIF_PROVIDER} \
-        --service-account=$${FAST_SERVICE_ACCOUNT} \
-        --service-account-token-lifetime-seconds=3600 \
-        --output-file=$${GOOGLE_CREDENTIALS} \
+        $FAST_WIF_PROVIDER \
+        --service-account=$FAST_SERVICE_ACCOUNT \
+        --service-account-token-lifetime-seconds=600 \
```
Comment (juliocc) [inline fast/assets/templates/workflow-gitlab.yaml:66]: 900, to make it consistent with the GH workflow?

File: fast/stages-multitenant/0-bootstrap-tenant/templates/workflow-gitlab.yaml [hunk 3]
```diff
@@ -12,113 +12,88 @@
...
-    - echo "$${GITLAB_TOKEN}" > token.txt
     - |
       gcloud iam workload-identity-pools create-cred-config \
-        $${FAST_WIF_PROVIDER} \
-        --service-account=$${FAST_SERVICE_ACCOUNT} \
-        --service-account-token-lifetime-seconds=3600 \
-        --output-file=$${GOOGLE_CREDENTIALS} \
+        $FAST_WIF_PROVIDER \
+        --service-account=$FAST_SERVICE_ACCOUNT \
+        --service-account-token-lifetime-seconds=600 \
```
Comment (juliocc) [inline fast/stages-multitenant/0-bootstrap-tenant/templates/workflow-gitlab.yaml:66]: idem

File: fast/stages-multitenant/1-resman-tenant/templates/workflow-gitlab.yaml [hunk 4]
```diff
@@ -12,113 +12,88 @@
...
-    - echo "$${GITLAB_TOKEN}" > token.txt
     - |
       gcloud iam workload-identity-pools create-cred-config \
-        $${FAST_WIF_PROVIDER} \
-        --service-account=$${FAST_SERVICE_ACCOUNT} \
-        --service-account-token-lifetime-seconds=3600 \
-        --output-file=$${GOOGLE_CREDENTIALS} \
+        $FAST_WIF_PROVIDER \
+        --service-account=$FAST_SERVICE_ACCOUNT \
+        --service-account-token-lifetime-seconds=600 \
```
Comment (juliocc) [inline fast/stages-multitenant/1-resman-tenant/templates/workflow-gitlab.yaml:66]: idem

File: fast/stages/0-bootstrap/templates/workflow-gitlab.yaml [hunk 5]
```diff
@@ -12,113 +12,88 @@
...
-    - echo "$${GITLAB_TOKEN}" > token.txt
     - |
       gcloud iam workload-identity-pools create-cred-config \
-        $${FAST_WIF_PROVIDER} \
-        --service-account=$${FAST_SERVICE_ACCOUNT} \
-        --service-account-token-lifetime-seconds=3600 \
-        --output-file=$${GOOGLE_CREDENTIALS} \
+        $FAST_WIF_PROVIDER \
+        --service-account=$FAST_SERVICE_ACCOUNT \
+        --service-account-token-lifetime-seconds=600 \
```
Comment (juliocc) [inline fast/stages/0-bootstrap/templates/workflow-gitlab.yaml:66]: idem

File: fast/stages/1-resman/templates/workflow-gitlab.yaml [hunk 6]
```diff
@@ -12,113 +12,88 @@
...
-    - echo "$${GITLAB_TOKEN}" > token.txt
     - |
       gcloud iam workload-identity-pools create-cred-config \
-        $${FAST_WIF_PROVIDER} \
-        --service-account=$${FAST_SERVICE_ACCOUNT} \
-        --service-account-token-lifetime-seconds=3600 \
-        --output-file=$${GOOGLE_CREDENTIALS} \
+        $FAST_WIF_PROVIDER \
+        --service-account=$FAST_SERVICE_ACCOUNT \
+        --service-account-token-lifetime-seconds=600 \
```
Comment (juliocc) [inline fast/stages/1-resman/templates/workflow-gitlab.yaml:66]: idem

**Review by juliocc — APPROVED** [review]
lgtm 🧑‍🎄

Left a couple of small comments

File: fast/assets/templates/workflow-github.yaml (hunk 1 above)
Comment (ludoo) [reply fast/assets/templates/workflow-github.yaml:81]: well, keep in mind this is fairly easy to exfiltrate, I would err on the side of caution :)

**Pull request merged**